modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ScriptPropertyDefValGenerator.cs

```diff
@@ -163,19 +164,81 @@ INamedTypeSymbol symbol
+                    .Select(r => r.GetSyntax() as PropertyDeclarationSyntax).FirstOrDefault();
 
                 // Fully qualify the value to avoid issues with namespaces.
                 string? value = null;
-                if (initializer != null)
+                if (propertyDeclarationSyntax?.Initializer != null)
                 {
-                    var sm = context.Compilation.GetSemanticModel(initializer.SyntaxTree);
-                    value = initializer.Value.FullQualifiedSyntax(sm);
+                    var sm = context.Compilation.GetSemanticModel(propertyDeclarationSyntax.Initializer.SyntaxTree);
+                    value = propertyDeclarationSyntax.Initializer.Value.FullQualifiedSyntax(sm);
+                }
+
+                if (string.IsNullOrWhiteSpace(value))
+                {
+                    var propertyGet = propertyDeclarationSyntax?.AccessorList?.Accessors.Where(a => a.Keyword.IsKind(SyntaxKind.GetKeyword)).FirstOrDefault();
+                    if (propertyGet != null)
+                    {
+                        if (propertyGet.ExpressionBody != null)
+                        {
+                            if (propertyGet.ExpressionBody.Expression is IdentifierNameSyntax identifierNameSyntax)
+                            {
+                                var fieldSymbol = members
+                                    .Where(s => s.Kind == SymbolKind.Field)
+                                    .Cast<IFieldSymbol>()
+                                    .Where(s => s.Name == identifierNameSyntax.Identifier.Text)
+                                    .FirstOrDefault();
+                                EqualsValueClauseSyntax? initializer = fieldSymbol?.DeclaringSyntaxReferences
+                                    .Select(r => r.GetSyntax())
+                                    .OfType<VariableDeclaratorSyntax>()
+                                    .Select(s => s.Initializer)
+                                    .FirstOrDefault(i => i != null);
+
+                                if (initializer != null)
+                                {
+                                    var sm = context.Compilation.GetSemanticModel(initializer.SyntaxTree);
+                                    value = initializer.Value.FullQualifiedSyntax(sm);
+                                }
+                            }
+                            else
+                            {//Expression not single field
+                                var sm = context.Compilation.GetSemanticModel(propertyGet.ExpressionBody.Expression.SyntaxTree);
+                                value = propertyGet.ExpressionBody.Expression.FullQualifiedSyntax(sm);
+                            }
+                        }
+                        else
+                        {
+                            var returnStatementSyntax = propertyGet.DescendantNodes().OfType<ReturnStatementSyntax>().FirstOrDefault();
```

Comment:
This doesn't make sure that the body of the getter is a single return statement, it only looks for that return statement. This can have many problems, even if the return example is just an identifier for a field. For example, a method could have more than one return statement.

IMO, the best we can support here is bodies that contain only a return statement with the identifier (as weird as it would be to write that with a block rather than an expression body). This code should make sure that's all the code does.

Reply:
only generate a single return, which should be applicable to most cases